lib/features/personal_summary/giving_goal/pages/setup_giving_goal_bottom_sheet.dart

```diff
@@ -0,0 +1,220 @@
+
+    return BottomSheetLayout(
+      title: Text(locals.budgetGivingGoalTitle),
+      bottomSheet: Column(
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          Visibility(
+            visible: personalSummaryState.givingGoal.monthlyGivingGoal != 0,
+            child: TextButton(
+              onPressed: () {
+                context
+                    .read<PersonalSummaryBloc>()
+                    .add(const PersonalSummaryGoalRemove());
+              },
+              child: Text(
+                locals.budgetGivingGoalRemove,
+                style: Theme.of(context).textTheme.bodyLarge!.copyWith(
+                      decoration: TextDecoration.underline,
+                      color: AppTheme.givtPurple,
+                    ),
+              ),
+            ),
+          ),
+          Visibility(
+            visible:
+                personalSummaryState.status == PersonalSummaryStatus.loading,
+            child: const Center(
+              child: CircularProgressIndicator(),
+            ),
+          ),
+          Visibility(
+            visible:
+                personalSummaryState.status != PersonalSummaryStatus.loading,
+            child: ElevatedButton(
+              onPressed: isEnabled ? () => onSave(context) : null,
+              style: ElevatedButton.styleFrom(
+                disabledBackgroundColor: Colors.grey,
+              ),
+              child: Text(
+                locals.save,
+              ),
+            ),
+          ),
+        ],
+      ),
+      child: BlocListener<PersonalSummaryBloc, PersonalSummaryState>(
+        listener: (context, state) {
+          if (state.status == PersonalSummaryStatus.noInternet) {
+            showDialog<void>(
+              context: context,
+              builder: (_) => WarningDialog(
+                title: locals.noInternetConnectionTitle,
+                content: locals.noInternet,
+                onConfirm: () => context.pop(),
+              ),
+            );
+          }
+
+          if (state.status == PersonalSummaryStatus.success) {
+            context.pop();
+          }
+        },
+        child: Column(
+          children: [
+            CardBanner(
+              title: locals.budgetGivingGoalInfoBold,
+              subtitle: locals.budgetGivingGoalInfo,
+            ),
+            _buildFieldTitle(locals.budgetGivingGoalMine),
+            _buildTextField(context, country),
+            _buildFieldTitle(locals.budgetGivingGoalTime),
+            _buildPeriodDropdown(context: context),
+          ],
+        ),
+      ),
+    );
+  }
+
+  /// Build the text field for the amount
+  Widget _buildTextField(
+    BuildContext context,
+    Country country,
+  ) {
+    final locals = context.l10n;
+    return TextFormField(
+      style: Theme.of(context).textTheme.bodyLarge,
+      controller: amountController,
+      inputFormatters: [
+        FilteringTextInputFormatter.allow(
```

Comment:
Looks like this is a copy of code that we already use. Should we start defining this inputFormatters somewhere? :)

Reply:
Done